### ISSUE TITLE
Recipe not found error after storage migration

### ISSUE THREAD

**Issue by GrimSqueaker** [conan-io/conan#4219]
I wanted to upgrade our Conan infrastructure from 1.2.3 to 1.11.1 which works well client-wise (i.e. 1.11.1 clients still connecting to the 1.2.3 server) but I ran into problems when updating the server.

The Conan server 1.2.3 is running on a openSUSE 15 via gunicorn in a virtualenv environment. For the update I have
- stopped the systemd service
- checked out 1.11.1 via git
- updated the PIP requirements
- restarted the systemd service

After that the service could not start and `systemctl status conan` yielded:
```
Jan 03 09:23:57 test gunicorn[3886]: EOF when reading a line
Jan 03 09:23:57 test gunicorn[3886]: **********************************************
Jan 03 09:23:57 test gunicorn[3886]: *                                            *
Jan 03 09:23:57 test gunicorn[3886]: *       STORAGE MIGRATION NEEDED!            *
Jan 03 09:23:57 test gunicorn[3886]: *                                            *
Jan 03 09:23:57 test gunicorn[3886]: **********************************************
Jan 03 09:23:57 test gunicorn[3886]: A migration of your storage is needed, please backup first the storage directory before continue:
Jan 03 09:23:57 test gunicorn[3886]: '/var/conan/data'
Jan 03 09:23:57 test gunicorn[3886]: Continue? Y/N An error ocurred during the migration, please restore the backup directory and try again
Jan 03 09:23:57 test gunicorn[3886]: EOF when reading a line
```
So, it is waiting for input from stdin, which gunicorn cannot provide.
I went ahead and wrote a script to launch the server on the command line without gunicorn
```
import sys
sys.path.append('/opt/conan/conan_repo')
from conans.server.server_launcher import main
main(sys.argv[1:])
```
and started it via
```
/opt/conan_virtenv/bin/python /opt/conan/conan_server.py
```
I was presented with the same migration question as before and answered with Y. The migration seemingly went well
```
[...]
**********************************************
*                                            *
*       MIGRATION COMPLETED!                 *
*                                            *
**********************************************
***********************
Using config: /home/conan/.conan_server/server.conf
Storage: /var/conan/data
Public URL: http://xxx.yyy:9342/v1
PORT: 9342
[...]
```
When I restart the server via systemd/gunicorn now, I can successfully perform searches from clients:
```
D:\temp\conantest>conan search -r test
Existing package recipes:
[...]
zlib/1.2.11@abc/testing
```
However, when I now want to retrieve a package or more information on the package, then I run into the "Recipe not found" error:
```
D:\temp\conantest>conan install -r test zlib/1.2.11@abc/testing
zlib/1.2.11@abc/testing: Not found, retrieving from server 'test'
zlib/1.2.11@abc/testing: Trying with 'test'...
ERROR: zlib/1.2.11@abc/testing was not found in remote 'test'

D:\temp\conantest>conan search -r test zlib/1.2.11@abc/testing
ERROR: Recipe not found: 'zlib/1.2.11@abc/testing'. [Remote: test]
```

When I check in `/var/conan/data` on the server, the directories have properly been moved to the revision sub directories "0".

**Did I miss a setting for the server s.t. either no migration is performed or after the migration the data is found correctly?**

Cheers,
GrimSqueaker

---

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by lasote** [comment]
Hi, thanks for the tip about gunicorn and stdin, we didn't take it into account, we will think what can we do about it.

About the errors after the migration, no setting adjustment is required to operate with the migrated data. I don't know what could be happening. Do you have some traces from the server?

Also, you could try to set `set CONAN_TRACE_FILE=c:\myfile.log` and then run conan again, to see the rest API calls failing against the server. `set CONAN_LOGGING_LEVEL=10` will also give some more traces on the client side.

**Comment by lasote** [comment]
@theodelrieu I think @memsharded told me you also had similar issues? What is the current status?

**Comment by GrimSqueaker** [comment]
Thanks for the quick reply.
On the client I get the following output:
```
D:\temp\conantest>conan search -r test zlib/1.2.11@abc/testing
DEBUG :conan_api.py   [202]: INIT: Using config 'C:\Users\sebastian.birk\.conan\conan.conf' [2019-01-03 16:02:50,331]
DEBUG :tracer.py      [151]: CONAN_API: search_packages(outdated=False,remote_name=test,query=None) [2019-01-03 16:02:50,346]
DEBUG :rest_client_common.py[133]: REST: ping: http://test:9342/v1/ping [2019-01-03 16:02:50,346]
DEBUG :rest_client_common.py[159]: REST: get: http://test:9342/v1/conans/zlib/1.2.11/abc/testing/search [2019-01-03 16:02:50,378]
DEBUG :rest_client_common.py[49]: REST ERROR: <class 'conans.errors.NotFoundException'> [2019-01-03 16:02:50,378]
ERROR: Recipe not found: 'zlib/1.2.11@abc/testing'. [Remote: test]
```
and in the trace file:
```
{"_action": "COMMAND", "name": "search_packages", "parameters": {"outdated": false, "query": null, "remote_name": "test"}, "time": 1546527770.3469963}
{"_action": "REST_API_CALL", "duration": 0.03119969367980957, "headers": {"Authorization": "**********", "User-Agent": "Conan/1.11.1 (Python 3.5.2) python-requests/2.18.4", "X-Client-Anonymous-Id": "**********", "X-Client-Id": "", "X-Conan-Client-Version": "1.11.1"}, "method": "GET", "time": 1546527770.378196, "url": "http://test:9342/v1/ping"}
{"_action": "REST_API_CALL", "duration": 0.0, "headers": {"Authorization": "**********", "User-Agent": "Conan/1.11.1 (Python 3.5.2) python-requests/2.18.4", "X-Client-Anonymous-Id": "**********", "X-Client-Id": "", "X-Conan-Client-Version": "1.11.1"}, "method": "GET", "time": 1546527770.378196, "url": "http://test:9342/v1/conans/zlib/1.2.11/abc/testing/search"}
{"_action": "EXCEPTION", "class": "NotFoundException", "message": "Recipe not found: 'zlib/1.2.11@abc/testing'. [Remote: test]", "time": 1546527770.378196}
```
I'll adjust the logging level on the server and check its logs too.

**Comment by lasote** [comment]
The requests look good. Could you paste the server storage folder representation for the recipe `zlib/1.2.11@abc/testing` ? Specially the `packages` folder: where is it, which folder contain...
For example, for my OpenSSL recipe I have:

```
~/.conan_server/data/OpenSSL/1.0.2o/conan/stable/0/package/75b541b6f810787a4fcee8cec9cfdbf95959366f/0

conaninfo.txt  conanmanifest.txt  conan_package.tgz
```

**Comment by GrimSqueaker** [comment]
Hmm, I cannot get anything useful to log on the server. When I set
```
Environment="CONAN_LOGGING_FILE=/home/conan/logging.log" "CONAN_TRACE_FILE=/home/conan/trace.log" "CONAN_LOGGING_LEVEL=10"
ExecStart=/opt/conan_virtenv/bin/gunicorn --pid /home/conan/run/gunicorn/pid -w 4 -t 420 --worker-tmp-dir /home/conan/temp --pythonpath '/opt/conan/conan_repo' --access-logfile '/var/conan/logs/gunicorn-access.log' --error-logfile '/var/conan/logs/gunicorn-error.log' --log-file '/var/conan/logs/gunicorn.log' conans.server.server_launcher:app
```
for the systemd service then I get
```
DEBUG :__init__.py    [83]: 'custom_authenticator' [2019-01-03 17:08:14,344]
DEBUG :__init__.py    [83]: 'custom_authenticator' [2019-01-03 17:08:14,428]
DEBUG :__init__.py    [83]: 'custom_authenticator' [2019-01-03 17:08:14,511]
DEBUG :__init__.py    [83]: 'custom_authenticator' [2019-01-03 17:08:14,517]
DEBUG :authorization_header.py[34]: Call: <function PingController.attach_to.<locals>.ping at 0x7ffacb925ea0> [2019-01-03 17:08:19,573]
DEBUG :authorization_header.py[34]: Call: <function PingController.attach_to.<locals>.ping at 0x7ffacb7d9f28> [2019-01-03 17:08:19,574]
DEBUG :authorization_header.py[34]: Call: <function SearchController.attach_to.<locals>.search_packages at 0x7ffacb934b70> [2019-01-03 17:08:19,579]
DEBUG :authorization_header.py[34]: Call: <function SearchController.attach_to.<locals>.search_packages at 0x7ffacb7df1e0> [2019-01-03 17:08:19,579]
```
in the log file and not trace or whatsoever.

**Comment by lasote** [comment]
everything looks good in the storage and the traces... maybe something wrong with the file contents? :(
Could you zip the zlib folder in the server data and send me? You can write me to luism@jfrog.com or lasote@gmail.com

**Comment by GrimSqueaker** [comment]
I think I found it. When I upload a local package I obtain a small difference on the server:
```
conan@dtest:/var/conan/data> find pugixml/
pugixml/
pugixml/1.8
pugixml/1.8/abc
pugixml/1.8/abc/thirdparty
pugixml/1.8/abc/thirdparty/revisions.txt.lock
pugixml/1.8/abc/thirdparty/revisions.txt
pugixml/1.8/abc/thirdparty/0
pugixml/1.8/abc/thirdparty/0/export
pugixml/1.8/abc/thirdparty/0/export/conanmanifest.txt
pugixml/1.8/abc/thirdparty/0/export/conanfile.py
pugixml/1.8/abc/thirdparty/0/package
pugixml/1.8/abc/thirdparty/0/package/928eed40bad56d27b26d67d062c83b9ffd6bc9aa
pugixml/1.8/abc/thirdparty/0/package/928eed40bad56d27b26d67d062c83b9ffd6bc9aa/revisions.txt.lock
pugixml/1.8/abc/thirdparty/0/package/928eed40bad56d27b26d67d062c83b9ffd6bc9aa/revisions.txt
pugixml/1.8/abc/thirdparty/0/package/928eed40bad56d27b26d67d062c83b9ffd6bc9aa/0
pugixml/1.8/abc/thirdparty/0/package/928eed40bad56d27b26d67d062c83b9ffd6bc9aa/0/conanmanifest.txt
pugixml/1.8/abc/thirdparty/0/package/928eed40bad56d27b26d67d062c83b9ffd6bc9aa/0/conaninfo.txt
pugixml/1.8/abc/thirdparty/0/package/928eed40bad56d27b26d67d062c83b9ffd6bc9aa/0/conan_package.tgz
pugixml/1.8/abc/thirdparty/0/package/f66c8ef68b7c0075b0922655535a5ef6cf38a460
pugixml/1.8/abc/thirdparty/0/package/f66c8ef68b7c0075b0922655535a5ef6cf38a460/revisions.txt.lock
pugixml/1.8/abc/thirdparty/0/package/f66c8ef68b7c0075b0922655535a5ef6cf38a460/revisions.txt
pugixml/1.8/abc/thirdparty/0/package/f66c8ef68b7c0075b0922655535a5ef6cf38a460/0
pugixml/1.8/abc/thirdparty/0/package/f66c8ef68b7c0075b0922655535a5ef6cf38a460/0/conanmanifest.txt
pugixml/1.8/abc/thirdparty/0/package/f66c8ef68b7c0075b0922655535a5ef6cf38a460/0/conaninfo.txt
pugixml/1.8/abc/thirdparty/0/package/f66c8ef68b7c0075b0922655535a5ef6cf38a460/0/conan_package.tgz
```
The migration did not create the **revisions.txt** and **revisions.txt.log** file.

Then for this particular recipe the search works fine:
```
D:\temp\conantest>conan search -r test pugixml/1.8@abc/thirdparty
DEBUG :conan_api.py   [202]: INIT: Using config 'C:\Users\sebastian.birk\.conan\conan.conf' [2019-01-04 08:50:15,467]
DEBUG :tracer.py      [151]: CONAN_API: search_packages(remote_name=test,outdated=False,query=None) [2019-01-04 08:50:15,477]
DEBUG :rest_client_common.py[133]: REST: ping: http://test:9342/v1/ping [2019-01-04 08:50:15,477]
DEBUG :rest_client_common.py[159]: REST: get: http://test:9342/v1/conans/pugixml/1.8/abc/thirdparty/search [2019-01-04 08:50:15,487]
DEBUG :rest_client_common.py[159]: REST: get: http://test:9342/v1/conans/pugixml/1.8/abc/thirdparty/digest [2019-01-04 08:50:15,497]
DEBUG :uploader_downloader.py[153]: DOWNLOAD: http://test:9342/v1/files/pugixml/1.8/abc/thirdparty/0/export/conanmanifest.txt?signature=[...] [2019-01-04 08:50:15,517]
Existing packages for recipe pugixml/1.8@abc/thirdparty:

Existing recipe in remote 'test':

    Package_ID: 928eed40bad56d27b26d67d062c83b9ffd6bc9aa
        [settings]
            arch: x86_64
            build_type: Debug
            compiler: Visual Studio
            compiler.runtime: MDd
            compiler.version: 11
            os: Windows
        Outdated from recipe: False

    Package_ID: f66c8ef68b7c0075b0922655535a5ef6cf38a460
        [settings]
            arch: x86_64
            build_type: Release
            compiler: Visual Studio
            compiler.runtime: MD
            compiler.version: 11
            os: Windows
        Outdated from recipe: False
```

Is this missing in the migration or has it somehow been screwed up on my system?

**Comment by lasote** [comment]
I think it is not necessary, these files are created on upload, but without them, "0" is assumed. Revisions are uploaded (still [blocked feature](https://github.com/conan-io/conan/wiki/Revisions-experimental-feature)). I have a package without these files and it is working good.
My only idea is to reproduce the issue and debug in the code what is happening, but I would need to reproduce it.

**Comment by lasote** [comment]
Thanks for the email, I reproduced and debugged the issue. You were right, the lack of the index files is making it break. Probably we will try to release ASAP a patched conan version that will create the indexes on the fly in this case.

**Comment by lasote** [comment]
Hi, the 1.11.2 release probably needs to wait until Thursday, does it blocks you? or you managed to restore the backup with an old server?

**Comment by theodelrieu** [comment]
Hi, we did remove every package eventually, and downgrade to 1.8.4... Happy to see that someone opened an issue!

**Comment by GrimSqueaker** [comment]
It doesn't block us either, because I have tested the migration on a non-productive system. Is there a problem if I use 1.11.1 clients against the 1.2.3 server?

**Comment by lasote** [comment]
No problem at all, thank a lot for your help and your patience. We will release asap next week.

**Comment by lasote** [comment]
This was released in 1.11.2
